app/src/main/java/com/hadiyarajesh/notex/ui/note/add/AddNoteScreen.kt

```diff
@@ -1,31 +1,190 @@
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
 import androidx.navigation.NavController
+import com.hadiyarajesh.notex.R
+import com.hadiyarajesh.notex.ui.component.BorderLessTextField
+import com.hadiyarajesh.notex.ui.component.VerticalSpacer
+import com.hadiyarajesh.notex.ui.note.add.NoteState.Companion.noteColors
+import kotlinx.coroutines.launch
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun AddNoteScreen(
-    navController: NavController
+    navController: NavController, addNotesViewModel: AddNotesViewModel, noteId: Long = -1
 ) {
-    Scaffold { innerPadding ->
+    val state = remember {
+        mutableStateOf(NoteState())
+    }
+
+    val noteBackground = remember {
+        Animatable(
+            Color(state.value.color.value)
+        )
+    }
+    LaunchedEffect(key1 = true) {
+        if (noteId.compareTo(-1) != 0) {
+            val note = addNotesViewModel.getNote(noteId)
+            state.value.apply {
+                title.value = note.title ?: ""
+                noteDesc.value = note.content ?: ""
+                color.value = Color(android.graphics.Color.parseColor(note.color)).toArgb()
+            }
+
+            noteBackground.animateTo(
+                targetValue = Color(state.value.color.value),
+                animationSpec = tween(
+                    durationMillis = 400
+                )
+            )
+
+        }
+    }
+
+    BackHandler {
+        addNotesViewModel.saveNote(state.value, noteId)
+        navController.navigateUp()
+    }
+
+    Scaffold(bottomBar = {
+        BottomAppBar(containerColor = noteBackground.value, modifier = Modifier.height(40.dp)) {
+            ShowColor(state = state, noteBackground)
+        }
+    }) { innerPadding ->
         Column(
             modifier = Modifier
                 .fillMaxSize()
+                .background(noteBackground.value)
                 .padding(innerPadding),
             horizontalAlignment = Alignment.CenterHorizontally,
-            verticalArrangement = Arrangement.Center
-        ) {
-            Text(text = "Add some notes here")
+            verticalArrangement = Arrangement.Top,
+
+            ) {
+            TopAppBar(navController, addNotesViewModel, state, noteId)
+            Column(
+                modifier = Modifier.padding(
+                    start = 16.dp,
+                    end = 16.dp,
+                    top = 10.dp,
+                    bottom = 1.dp
+                )
+            ) {
+                BorderLessTextField(
+                    text = state.value.title.value,
+                    hint = "Title",
```

Comment:
please use `stringResource()` instead of hard-coded strings

Reply:
okay